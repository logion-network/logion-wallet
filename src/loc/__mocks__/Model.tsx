import {
    rejectProtectionRequest,
    acceptProtectionRequest,
} from './ModelMock';

export {
    rejectProtectionRequest,
    acceptProtectionRequest,
};
