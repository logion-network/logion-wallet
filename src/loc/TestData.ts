import { LegalOfficerCase, UUID, Hash } from "@logion/node-api";
import { LocData, HashString } from "@logion/client";

export function buildLocRequest(locId: UUID, loc: LegalOfficerCase): LocData {
    return {
        ownerAddress: loc.owner,
        requesterAddress: loc.requesterAddress,
        requesterLocId: loc.requesterLocId,
        description: "Description",
        locType: loc.locType,
        createdOn: "",
        decisionOn: "",
        id: locId,
        status: loc.closed ? "CLOSED" : "OPEN",
        files: loc.files.map((locFile, index) => ({
            ...locFile,
            nature: HashString.fromValue(`File ${index}`),
            name: `File ${index}`,
            addedOn: "",
            published: false,
            restrictedDelivery: false,
            contentType: "text/plain",
            size: 42n,
            status: "DRAFT",
        })),
        metadata: loc.metadata.map((locFile, index) => ({
            ...locFile,
            name: HashString.fromValue(`Data ${index}`),
            value: HashString.fromValue(`Value ${index}`),
            addedOn: "",
            published: false,
            status: "DRAFT",
        })),
        links: loc.links.map((locFile, index) => ({
            ...locFile,
            name: `Link ${index}`,
            nature: HashString.fromValue(`Nature ${index}`),
            addedOn: "",
            target: locFile.id.toString(),
            published: false,
            status: "DRAFT",
        })),
        closed: false,
        verifiedIssuer: false,
        issuers: [],
    };
}
