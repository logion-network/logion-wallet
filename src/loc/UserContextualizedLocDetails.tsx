import { LegalOfficer, UploadableCollectionItem, LocData, ClosedCollectionLoc } from "@logion/client";
import { useCallback, useEffect, useState } from "react";
import { useNavigate } from "react-router";
import { Col, OverlayTrigger } from "react-bootstrap";
import Tooltip from 'react-bootstrap/Tooltip';

import { useCommonContext } from "../common/CommonContext";
import { FullWidthPane } from "../common/Dashboard";
import Tabs from "../common/Tabs";
import { Row } from "../common/Grid";
import { POLKADOT, RED, BackgroundAndForegroundColors, BLUE } from "../common/ColorTheme";
import Icon from "../common/Icon";
import DangerFrame from "../common/DangerFrame";
import NewTabLink from "../common/NewTabLink";
import InlineDateTime from "../common/InlineDateTime";
import IconTextRow from "../common/IconTextRow";
import { useLogionChain } from "../logion-chain";
import { format } from "../common/DateTimeFormat";

import { UserLocPrivateFileButton } from "./LocPrivateFileButton";
import CheckFileFrame, { DocumentCheckResult } from 'src/components/checkfileframe/CheckFileFrame';
import { UserLocPublicDataButton } from "./LocPublicDataButton";
import { UserLocItems } from "./LocItems";
import LocItemDetail from "./LocItemDetail";
import CertificateAndLimits from "./CertificateAndLimits";
import { UserCollectionLocItemChecker } from "./CollectionLocItemChecker";
import ItemImporter from "./ItemImporter";
import { useUserLocContext } from "./UserLocContext";

import "./ContextualizedLocDetails.css";

export default function UserContextualizedLocDetails() {
    const { getOfficer } = useLogionChain();
    const { colorTheme } = useCommonContext();
    const navigate = useNavigate();
    const { locState, loc, locId, locItems, backPath, detailsPath } = useUserLocContext();
    const [ checkResult, setCheckResult ] = useState<DocumentCheckResult>({result: "NONE"});
    const [ collectionItem, setCollectionItem ] = useState<UploadableCollectionItem>();
    const [ legalOfficer, setLegalOfficer ] = useState<LegalOfficer | null>(null)
    const [ supersededLoc, setSupersededLoc ] = useState<LocData | undefined | null>(null)

    useEffect(() => {
        if (loc && loc.replacerOf && locState && supersededLoc === null) {
            locState.publicSupersededLoc()
                .then(setSupersededLoc)
        }
    }, [ loc, locState, supersededLoc ])

    const checkHash = useCallback(async (hash: string) => {
        setCollectionItem(undefined);

        for(let i = 0; i < locItems!.length; ++i) {
            if(locItems[i].type === "Document" || locItems[i].type === "Data") {
                const file = locItems[i];
                if(file.value === hash) {
                    setCheckResult({
                        result: "POSITIVE",
                        hash
                    });
                    return;
                }
            }
        }

        if (locState instanceof ClosedCollectionLoc) {
            const collectionItem = (await locState.checkHash(hash)).collectionItem;
            if (collectionItem) {
                setCollectionItem(collectionItem);
                setCheckResult({
                    result: "POSITIVE",
                    hash
                });
                return;
            }
        }

        setCheckResult({
            result: "NEGATIVE",
            hash
        });
        setCollectionItem(undefined);
    }, [ locItems, setCheckResult, setCollectionItem, locState ]);

    useEffect(() => {
        if (legalOfficer === null && loc) {
            const owner = getOfficer!(loc.ownerAddress);
            if(owner !== undefined) {
                setLegalOfficer(owner);
            }
        }
    }, [ legalOfficer, loc, getOfficer, setLegalOfficer ])

    if (loc === null) {
        return null;
    }

    let locTabBorderColor = BLUE;
    if(loc.voidInfo !== undefined) {
        locTabBorderColor = RED;
    } else if(loc.closed) {
        locTabBorderColor = POLKADOT;
    }

    let locTabBorderWidth: string | undefined = undefined;
    if(loc.voidInfo !== undefined) {
        locTabBorderWidth = "2px";
    }

    let tabColors: BackgroundAndForegroundColors | undefined = undefined;
    if(loc.voidInfo !== undefined) {
        tabColors = {
            foreground: "white",
            background: RED
        };
    }

    let paneTitle: string = "";
    let paneIcon: string = "";
    if(loc.locType === 'Transaction') {
        paneTitle = "Transaction Protection Case";
        paneIcon = 'loc';
    } else if(loc.locType === 'Identity') {
        paneTitle = "Identity Case";
        paneIcon = 'identity';
    } else if(loc.locType === 'Collection') {
        paneTitle = "Collection Protection Case";
        paneIcon = 'collection';
    }

    let locTabTitle: string;
    if(loc.locType === 'Transaction') {
        locTabTitle = "Legal Officer Case (LOC) - Transaction";
    } else if(loc.locType === 'Collection') {
        locTabTitle = "Legal Officer Case (LOC) - Collection";
    } else {
        if (locState?.isLogionIdentity()) {
            locTabTitle = "Legal Officer Case (LOC) - Logion Identity";
        } else {
            locTabTitle = "Legal Officer Case (LOC) - Polkadot Identity";
        }
    }
    if(loc.voidInfo !== undefined) {
        locTabTitle = "VOID " + locTabTitle;
    }

    return (
        <FullWidthPane
            mainTitle={ paneTitle }
            titleIcon={ {
                icon: {
                    id: paneIcon
                },
                background: colorTheme.topMenuItems.iconGradient,
            } }
            onBack={ () => navigate(backPath) }
            className="ContextualizedLocDetails"
        >
            <Tabs
                id="loc-content"
                activeKey="details"
                onSelect={ () => {
                } }
                tabs={ [ {
                    key: "details",
                    title: locTabTitle,
                    render: () => {
                        const { date, time } = format(loc.createdOn);
                        let closingDate: string;
                        if (loc.closedOn !== undefined) {
                            const { date, time } = format(loc.closedOn);
                            closingDate = `${ date } / ${ time }`;
                        } else {
                            closingDate = "";
                        }
                        return <>
                            <Row>
                                <Col md={ 4 }>
                                    <LocItemDetail label="LOC ID" copyButtonText={ locId.toDecimalString() } >
                                        <OverlayTrigger
                                            placement="top"
                                            delay={ 500 }
                                            overlay={
                                                <Tooltip id={ locId.toDecimalString() }>{ locId.toDecimalString() }</Tooltip> }>
                                            <span>{ locId.toDecimalString() }</span>
                                        </OverlayTrigger>
                                    </LocItemDetail>
                                    <LocItemDetail label="Creation date">{ date } / { time }</LocItemDetail>
                                </Col>
                                <Col md={ 4 }>
                                    <LocItemDetail label="Description">{ loc?.description }</LocItemDetail>
                                    {
                                        loc.status === 'CLOSED' &&
                                        <LocItemDetail label="Closing date" spinner={ loc.closedOn === undefined }>{ closingDate }</LocItemDetail>
                                    }
                                </Col>

                                <Col md={ 4 } className="closed-icon-container">
                                        <LocItemDetail label="Legal Officer in charge">
                                            { legalOfficer?.name || "" }
                                            <OverlayTrigger
                                                placement="top"
                                                delay={ 500 }
                                                overlay={
                                                    <Tooltip
                                                        id={ loc?.ownerAddress }>{ loc?.ownerAddress }</Tooltip> }>
                                                <span><br /> { loc?.ownerAddress }</span>
                                            </OverlayTrigger>
                                        </LocItemDetail>
                                    {
                                        loc.closed && loc.voidInfo === undefined &&
                                        <div className="closed-icon">
                                            <Icon icon={ { id: "polkadot_shield" } } />
                                        </div>
                                    }
                                    {
                                        loc.voidInfo !== undefined &&
                                        <div className="closed-icon">
                                            <Icon icon={ { id: "void_shield" } } />
                                        </div>
                                    }
                                </Col>
                            </Row>
                            <div className="separator" style={{ backgroundColor: locTabBorderColor }} />
                            <UserLocItems matchedHash={ checkResult.hash } />
                            {
                                !loc.closed && loc.voidInfo === undefined &&
                                <Row>
                                    <Col className="add-buttons-container" xxl={5} xl={4}>
                                        <UserLocPublicDataButton/>
                                        <UserLocPrivateFileButton/>
                                    </Col>
                                    <Col className="link-button-container" xxl={4} xl={4}/>
                                    <Col className="close-button-container" xxl={3} xl={4}/>
                                </Row>
                            }
                        </>
                    }
                } ] }
                borderColor={ locTabBorderColor }
                borderWidth={ locTabBorderWidth }
                tabColors={ tabColors }
                flatBottom={ loc.voidInfo !== undefined }
            />
            {
                loc.voidInfo !== undefined && loc.locType !== 'Collection' &&
                <DangerFrame
                    className="loc-is-void"
                    title={ <span><Icon icon={ { id: 'void' } } width="45px" /> This LOC is VOID</span> }
                >
                    <p><strong>You have voided this LOC at the following date:</strong> <InlineDateTime dateTime={ loc.voidInfo?.voidedOn } /></p>
                    <p><strong>Reason:</strong> { loc.voidInfo?.reason || "-" }</p>
                    {
                        loc.voidInfo.replacer !== undefined &&
                        <p><strong>This VOID LOC has been replaced by the following LOC: </strong>
                            <NewTabLink
                                href={ detailsPath(loc.voidInfo.replacer, loc.locType) }
                                iconId="loc-link"
                                inline
                            >
                                { loc.voidInfo.replacer.toDecimalString() }
                            </NewTabLink>
                        </p>
                    }
                    {
                        loc.voidInfo.replacer === undefined &&
                        <p>Please note that its public certificate shows a "VOID" mention to warn people that the content of the LOC is not valid anymore.</p>
                    }
                    {
                        loc.voidInfo.replacer !== undefined &&
                        <p>Please note that its public certificate shows a "VOID" mention to warn people that the content of the LOC is not valid anymore.
                            People will be automatically redirected to the replacing LOC when accessing to the void LOC URL and a mention of the fact that
                            the replacing LOC supersedes the void LOC will be visible on both certificates.
                        </p>
                    }
                </DangerFrame>
            }
            {
                loc.voidInfo !== undefined && loc.locType === 'Collection' &&
                <DangerFrame
                    className="loc-is-void"
                    title={ <span><Icon icon={ { id: 'void' } } width="45px" /> This Collection LOC with all its related Collection Items are VOID</span> }
                >
                    <p><strong>You have voided this Collection LOC with all its related Collection Items at the following date:</strong> <InlineDateTime dateTime={ loc.voidInfo?.voidedOn } /></p>
                    <p><strong>Reason:</strong> { loc.voidInfo?.reason || "-" }</p>
                    <p>Please note that related public certificates show a "VOID" mention to warn people that the content of the Collection LOC as well as its related Collection Items are not valid anymore.</p>
                </DangerFrame>
            }
            <CertificateAndLimits
                locId={ locId }
                loc={{ ...loc, isVoid: loc.voidInfo !== undefined }}
                viewer="User"
            />
            { loc.locType === 'Collection' && loc.closed &&
                <UserCollectionLocItemChecker
                    locId={ locId }
                    locOwner={ loc.ownerAddress }
                    collectionItem={ collectionItem }
                />
            }
            { loc.locType === 'Collection' && loc.closed && loc.voidInfo === undefined &&
                <ItemImporter
                    locId={ locId }
                    collectionItem={ collectionItem }
                />
            }
            {
                supersededLoc !== null && supersededLoc !== undefined &&
                <DangerFrame
                    className="loc-supersedes"
                >
                    <IconTextRow
                        icon={ <Icon icon={ { id: 'void_supersede' } } width="45px" /> }
                        text={
                            <>
                                <p className="frame-title">IMPORTANT: this logion Legal Officer Case (LOC) supersedes a previous LOC (VOID)</p>
                                <p><strong>This LOC supersedes a previous LOC (VOID) since the following date:</strong> <InlineDateTime dateTime={ supersededLoc.voidInfo?.voidedOn } /></p>
                                <p><strong>For record purpose, this LOC supersedes the following LOC: </strong>
                                    <NewTabLink
                                        href={ detailsPath(loc.replacerOf!, loc.locType) }
                                        iconId="loc-link"
                                        inline
                                    >
                                        { loc.replacerOf!.toDecimalString() }
                                    </NewTabLink>
                                </p>
                            </>
                        }
                    />
                </DangerFrame>
            }
            <CheckFileFrame
                checkHash={ checkHash }
                checkResult={ checkResult.result }
            />
        </FullWidthPane>
    );
}
