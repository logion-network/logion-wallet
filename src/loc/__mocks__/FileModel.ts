export { getAllDeliveries, getAllCollectionDeliveries } from './FileModelMock';
