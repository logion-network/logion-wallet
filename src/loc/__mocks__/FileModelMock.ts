export let getAllDeliveries = jest.fn().mockResolvedValue({});
