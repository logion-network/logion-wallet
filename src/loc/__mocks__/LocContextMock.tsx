import { LocData, CollectionItem } from "@logion/client";
import { UUID, ValidAccountId } from "@logion/node-api";

import { LocItem } from "../LocItem";

let locState: any = {
    isLogionIdentity: () => false,
    isLogionData: () => true,
};

export function setLocState(value: any) {
    locState = value;
    if(value.data) {
        loc = {
            ...loc,
            ...locState.data(),
        }
    }
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

let locId = new UUID("aed4c6e4-979e-48ad-be6e-4bd39fb94762");

export function setLocId(id: UUID) {
    locId = id;
}

let loc: Partial<LocData> | null = {
    id: locId,
    locType: "Identity",
    requesterAccountId: ValidAccountId.polkadot("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
        email: "john.doe@logion.network",
        phoneNumber: "+1234",
    }
};

export function setLocRequest(request: Partial<LocData> | null) {
    loc = request;
}

export let refresh = jest.fn();

export let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

export let deleteMetadata = jest.fn();

export let deleteLink = jest.fn();

let collectionItems: CollectionItem[] = [];

export function setCollectionItems(items: CollectionItem[]) {
    collectionItems = items;
}

async function mutateLocState(mutator: (state: any) => Promise<any>): Promise<void> {
    await mutator(locState);
}

export function useLocContext() {
    return {
        linkLoc: {

        },
        locId,
        loc,
        locItems,
        refresh,
        requestSof,
        deleteMetadata,
        deleteLink,
        checkResult: { result: "NONE" },
        locState,
        collectionItems,
        mutateLocState,
    };
}
