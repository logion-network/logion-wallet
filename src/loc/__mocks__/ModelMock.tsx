export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let acceptLocRequest = jest.fn();

export function setAcceptLocRequest(mockFn: any) {
    acceptLocRequest = mockFn;
}

export let rejectLocRequest = jest.fn();

export function setRejectLocRequest(mockFn: any) {
    rejectLocRequest = mockFn;
}

export let addLink = jest.fn();
