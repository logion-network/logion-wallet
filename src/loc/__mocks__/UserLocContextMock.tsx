import { UUID } from "@logion/node-api/dist/UUID";
import { LocData } from "@logion/client/dist/Loc";

import { LocRequest } from "../../common/types/ModelTypes";
import { LocItem } from "../types";

let locState: any;

export function setLocState(value: any) {
    locState = value;
}

export let refresh = jest.fn();

let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

let requestSofOnCollection = jest.fn();

export function setRequestSofOnCollection(fn: jest.Mock) {
    requestSofOnCollection = fn;
}

let locId: UUID;

export function setLocId(id: UUID) {
    locId = id;
}

let loc: LocData | null = null;

export function setLoc(request: LocData | null) {
    loc = request;
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

export let deleteMetadata = jest.fn();

export let deleteFile = jest.fn();

export let addFile = jest.fn().mockResolvedValue(undefined);

export let addMetadata = jest.fn().mockResolvedValue(undefined);

export function useUserLocContext() {

    return {
        refresh,
        locState,
        requestSof,
        requestSofOnCollection,
        locId,
        loc,
        deleteMetadata,
        deleteFile,
        locItems,
        addFile,
        addMetadata,
    }
}
