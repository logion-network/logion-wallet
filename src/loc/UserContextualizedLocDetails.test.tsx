import { UUID } from "@logion/node-api";
import { OPEN_IDENTITY_LOC, OPEN_IDENTITY_LOC_ID } from "../__mocks__/LogionMock";
import { shallowRender } from "../tests";

import UserContextualizedLocDetails from "./UserContextualizedLocDetails"
import { buildLocRequest } from "./TestData";
import { setLoc, setLocId } from "./__mocks__/UserLocContextMock";

jest.mock("../logion-chain/");
jest.mock("./UserLocContext");

describe("UserContextualizedLocDetails", () => {

    it("renders for requester", () => {
        const uuid = UUID.fromDecimalString(OPEN_IDENTITY_LOC_ID)!;
        setLocId(uuid);
        setLoc(buildLocRequest(uuid, OPEN_IDENTITY_LOC));
        const tree = shallowRender(<UserContextualizedLocDetails contributionMode="Requester" />);
        expect(tree).toMatchSnapshot();
    })

    it("renders for VTP", () => {
        const uuid = UUID.fromDecimalString(OPEN_IDENTITY_LOC_ID)!;
        setLocId(uuid);
        setLoc(buildLocRequest(uuid, OPEN_IDENTITY_LOC));
        const tree = shallowRender(<UserContextualizedLocDetails contributionMode="VTP" />);
        expect(tree).toMatchSnapshot();
    })
})
