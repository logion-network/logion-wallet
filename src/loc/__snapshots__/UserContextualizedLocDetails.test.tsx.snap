// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContextualizedLocDetails renders 1`] = `
<LocPane
  loc={
    Object {
      "closed": false,
      "createdOn": "",
      "decisionOn": "",
      "description": "Description",
      "files": Array [],
      "id": UUID {
        "bytes": Array [
          147,
          99,
          195,
          214,
          209,
          7,
          66,
          26,
          164,
          75,
          133,
          199,
          250,
          176,
          184,
          67,
        ],
      },
      "links": Array [],
      "locType": "Identity",
      "metadata": Array [],
      "ownerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "requesterAddress": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
      "requesterLocId": undefined,
      "status": "OPEN",
    }
  }
>
  <LocDetailsTab
    addFile={[MockFunction]}
    addMetadata={[MockFunction]}
    checkResult={
      Object {
        "result": "NONE",
      }
    }
    deleteFile={[MockFunction]}
    deleteLink={null}
    deleteMetadata={[MockFunction]}
    loc={
      Object {
        "closed": false,
        "createdOn": "",
        "decisionOn": "",
        "description": "Description",
        "files": Array [],
        "id": UUID {
          "bytes": Array [
            147,
            99,
            195,
            214,
            209,
            7,
            66,
            26,
            164,
            75,
            133,
            199,
            250,
            176,
            184,
            67,
          ],
        },
        "links": Array [],
        "locType": "Identity",
        "metadata": Array [],
        "ownerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "requesterAddress": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        "requesterLocId": undefined,
        "status": "OPEN",
      }
    }
    locItems={Array []}
    viewer="User"
  />
  <VoidDisclaimer
    loc={
      Object {
        "closed": false,
        "createdOn": "",
        "decisionOn": "",
        "description": "Description",
        "files": Array [],
        "id": UUID {
          "bytes": Array [
            147,
            99,
            195,
            214,
            209,
            7,
            66,
            26,
            164,
            75,
            133,
            199,
            250,
            176,
            184,
            67,
          ],
        },
        "links": Array [],
        "locType": "Identity",
        "metadata": Array [],
        "ownerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "requesterAddress": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        "requesterLocId": undefined,
        "status": "OPEN",
      }
    }
  />
  <CertificateAndLimits
    loc={
      Object {
        "closed": false,
        "createdOn": "",
        "decisionOn": "",
        "description": "Description",
        "files": Array [],
        "id": UUID {
          "bytes": Array [
            147,
            99,
            195,
            214,
            209,
            7,
            66,
            26,
            164,
            75,
            133,
            199,
            250,
            176,
            184,
            67,
          ],
        },
        "links": Array [],
        "locType": "Identity",
        "metadata": Array [],
        "ownerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "requesterAddress": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        "requesterLocId": undefined,
        "status": "OPEN",
      }
    }
    viewer="User"
  />
  <SupersedesDisclaimer
    loc={
      Object {
        "closed": false,
        "createdOn": "",
        "decisionOn": "",
        "description": "Description",
        "files": Array [],
        "id": UUID {
          "bytes": Array [
            147,
            99,
            195,
            214,
            209,
            7,
            66,
            26,
            164,
            75,
            133,
            199,
            250,
            176,
            184,
            67,
          ],
        },
        "links": Array [],
        "locType": "Identity",
        "metadata": Array [],
        "ownerAddress": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        "requesterAddress": "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        "requesterLocId": undefined,
        "status": "OPEN",
      }
    }
  />
  <CheckFileFrame
    checkResult="NONE"
    checkedItem="confidential document"
    context="Identity LOC"
  />
</LocPane>
`;
